src/tests/Microsoft.PowerFx.Core.Tests.Shared/ExpressionTestCases/Join.txt

```diff
@@ -0,0 +1,81 @@
+﻿#SETUP: PowerFxV1CompatibilityRules,StronglyTypedBuiltinEnums
+
+>> Join(
+  Table({Id:"AAA111",Name:"Contoso",Country:"USA"},{Id:"BBB222",Name:"Fabrikam",Country:"CAN"},{Id:"EEE555",Name:"ACME INC.",Country:"MEX"},{Id:"FFF666",Name:"Dunder Mifflin",Country:"USA"}),
+  Table({SupplierId:"AAA111", Fruit:"Grapes", Price:220, Purchase:Date(2015,10,1), RowId:1},
+  {SupplierId:"BBB222", Fruit:"Lemons", Price:31, Purchase:Date(2015,10,1), RowId:2},
+  {SupplierId:"AAA111", Fruit:"Lemons", Price:29, Purchase:Date(2015,10,2), RowId:3},
+  {SupplierId:"AAA111", Fruit:"Grapes", Price:210, Purchase:Date(2015,10,2), RowId:4},
+  {SupplierId:"BBB222", Fruit:"Lemons", Price:30, Purchase:Date(2015,10,3), RowId:5},
+  {SupplierId:"AAA111", Fruit:"Bananas", Price:12, Purchase:Date(2015,10,3), RowId:6},
+  {SupplierId:"CCC333", Fruit:"Apples", Price:15, Purchase:Date(2015,12,3), RowId:7},
+  {SupplierId:"DDD444", Fruit:"Mangos", Price:30, Purchase:Date(2016,12,3), RowId:8}),
+  LeftRecord.Id = RightRecord.SupplierId, JoinType.Inner)
```

Comment:
>JoinType.Inner [](http://example.com/codeflow?start=42&length=14)

Does this need to be a constant? 
(it would be ideal to restrict it to that - we can always lift the restriction later) 

Ie, fail if
If(true, JoinType.Inner, JoinType.Left)

Reply:
Starting out as a compile time constant is probably a good idea - it may help with delegation.

Related, JoinType should be a strongly typed enum that can't be combined.